cmd/flags_misc.go

```diff
@@ -201,5 +208,17 @@ func ParseMiscFlags(ctx context.Context, cmd *cobra.Command) (context.Context, e
 		ctx = WithNodeOptions(ctx, nodebuilder.WithMetrics(opts, NodeType(ctx)))
 	}
 
+	ok, err = cmd.Flags().GetBool(libp2pMetrics)
+	if err != nil {
+		panic(err)
+	}
+
+	if ok {
+		if metricsEnabled, _ := cmd.Flags().GetBool(metricsFlag); !metricsEnabled {
+			panic(fmt.Errorf("--libp2p.metrics requires --metrics to be enabled"))
```

Comment:
Just a regular error return instead of panic? users don't like seeing stacktraces